app/assets/src/components/views/report/SampleDetailsSidebar/MetadataEditor.jsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import Input from "~/components/ui/controls/Input";
+import Dropdown from "~/components/ui/controls/dropdowns/Dropdown";
+import cs from "./sample_details_sidebar.scss";
+import { isArray } from "lodash";
+import PropTypes from "prop-types";
+
+class MetadataEditor extends React.Component {
+  renderInput = metadataType => {
+    const { metadata, onMetadataChange, onMetadataSave } = this.props;
+
+    if (isArray(metadataType.options)) {
+      const options = metadataType.options.map(option => ({
+        text: option,
+        value: option
+      }));
+      return (
+        <Dropdown
+          fluid
+          floating
+          scrolling
+          options={options}
+          onChange={val => onMetadataChange(metadataType.key, val, true)}
+          value={metadata[metadataType.key]}
+        />
+      );
+    }
+
+    return (
+      <Input
+        onChange={val => onMetadataChange(metadataType.key, val)}
+        onBlur={() => onMetadataSave(metadataType.key)}
+        value={metadata[metadataType.key]}
+        type={metadataType.dataType === "number" ? "number" : "text"}
+        className={cs.input}
+      />
+    );
+  };
+
+  render() {
+    const { metadataTypes } = this.props;
+    return (
+      <div>
+        {metadataTypes.map(metadataType => (
+          <div className={cs.field} key={metadataType.key}>
+            <div className={cs.label}>{metadataType.name}</div>
+            {this.renderInput(metadataType)}
+          </div>
+        ))}
+      </div>
+    );
+  }
+}
+
+MetadataEditor.propTypes = {
+  metadata: PropTypes.shape({
```

Comment:
Why not add these types for `utils/propTypes`?
We can defer it to when we want to use it again, but they seem very common.

Reply:
Actually you can accept here more than just collection_location, no?